File: app/src/test/java/no/nav/tilbakemeldingsmottak/validators/SendRosValidatorTest.java
package no.nav.tilbakemeldingsmottak.validators;

import no.nav.tilbakemeldingsmottak.exceptions.ClientErrorException;
import no.nav.tilbakemeldingsmottak.model.SendRosRequest;
import no.nav.tilbakemeldingsmottak.rest.ros.validation.SendRosValidator;
import org.junit.jupiter.api.Test;

import static no.nav.tilbakemeldingsmottak.TestUtils.createSendRosRequest;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

class SendRosValidatorTest {

    private final SendRosValidator sendRosValidator = new SendRosValidator();
    private SendRosRequest sendRosRequest;

    @Test
    void happyPath() {
        sendRosRequest = createSendRosRequest();
        sendRosValidator.validateRequest(sendRosRequest);
    }

    @Test
    void shouldThrowExceptionIfHvemRosesNotSet() {
        sendRosRequest = createSendRosRequest();
        sendRosRequest.setHvemRoses(null);
        Exception thrown = assertThrows(ClientErrorException.class,
                () -> sendRosValidator.validateRequest(sendRosRequest));
        assertTrue(thrown.getMessage().contains("hvemRoses er påkrevd"));
    }

    @Test
    void shouldThrowExceptionIfKontorNotSet() {
        sendRosRequest = createSendRosRequest();
        sendRosRequest.setHvemRoses(SendRosRequest.HvemRosesEnum.NAV_KONTOR);
        Exception thrown = assertThrows(ClientErrorException.class,
                () -> sendRosValidator.validateRequest(sendRosRequest));
        assertTrue(thrown.getMessage().contains("navKontor er påkrevd dersom hvemRoses=NAV_KONTOR"));
    }

    @Test
    void shouldThrowExceptionIfMeldingNotSet() {
        sendRosRequest = createSendRosRequest();
        sendRosRequest.setMelding(null);
        Exception thrown = assertThrows(ClientErrorException.class,
                () -> sendRosValidator.validateRequest(sendRosRequest));
        assertTrue(thrown.getMessage().contains("melding er påkrevd"));
    }

}

File: app/src/test/kotlin/no/nav/tilbakemeldingsmottak/validators/SendRosValidatorTest.kt
package no.nav.tilbakemeldingsmottak.validators

import no.nav.tilbakemeldingsmottak.exceptions.ClientErrorException
import no.nav.tilbakemeldingsmottak.model.SendRosRequest
import no.nav.tilbakemeldingsmottak.model.SendRosRequest.HvemRoses
import no.nav.tilbakemeldingsmottak.rest.ros.validation.SendRosValidator
import no.nav.tilbakemeldingsmottak.util.builders.SendRosRequestBuilder
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

internal class SendRosValidatorTest {
    private val sendRosValidator = SendRosValidator()
    private var sendRosRequest: SendRosRequest = SendRosRequestBuilder().build()

    @Test
    fun happyPath() {
        sendRosRequest = SendRosRequestBuilder().build()
        sendRosValidator.validateRequest(sendRosRequest)
    }

    @Test
    fun shouldThrowExceptionIfHvemRosesNotSet() {
        // Given
        sendRosRequest = SendRosRequestBuilder().build(hvemRoses = null)

        // When
        val thrown =
            Assertions.assertThrows(ClientErrorException::class.java) { sendRosValidator.validateRequest(sendRosRequest) }

        // Then
        assertTrue(thrown.message.contains("hvemRoses er påkrevd"))
    }

    @Test
    fun shouldThrowExceptionIfKontorNotSet() {
        // Given
        sendRosRequest = SendRosRequestBuilder().build(hvemRoses = HvemRoses.NAV_KONTOR)

        // When
        val thrown =
            Assertions.assertThrows(ClientErrorException::class.java) { sendRosValidator.validateRequest(sendRosRequest) }

        // Then
        assertTrue(thrown.message.contains("navKontor er påkrevd dersom hvemRoses=NAV_KONTOR"))
    }

    @Test
    fun shouldThrowExceptionIfMeldingNotSet() {
        // Given
        sendRosRequest = SendRosRequestBuilder().build(melding = null)

        // When
        val thrown =
            Assertions.assertThrows(ClientErrorException::class.java) { sendRosValidator.validateRequest(sendRosRequest) }

        // Then
        assertTrue(thrown.message.contains("melding er påkrevd"))
    }
}